### PR TITLE
Add the retrieve_temporary_list member to the CalcInfo datastructure

### LINKED ISSUE #886: Introduce concept of temporary_retrieve_file_list to Parser class
A common use case for `Calculcation` and `Parser` implementations might be that the calculation produces large files that one does not want to store permanently in the `RetrievedFolder` but it still contains valuable information that the user wants to parse and then store in output nodes. Currently, one has to choose to either not retrieve the files and hence not store the desired information or parse the information after the fact directly from the `RemoteFolder` by using something like an `InlineCalculation`.

An example would be to parse occupation numbers for a `PwCalculation`. Ideally one would parse these from the xml files, but they are large and otherwise pretty useless, so we do not want to store these by default. The current alternative is to parse them from the `RemoteFolder` with the `get_bands_and_occupations_inline` function.

If we were to introduce another file list in the `AbstractCalculation` class `temporary_retrieve_list` in which an implementation can specify certain remote files that upon retrieval will be stored locally temporarily, for the duration of the parsing call. After the parsing is done, be it successful or not, these temporary files will be deleted. During the parsing they will then be available for parsing just like normal retrieved files.

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #886

### CODE REVIEW

File: aiida/common/datastructures.py [hunk 1]
```diff
@@ -114,24 +114,25 @@ class CalcInfo(DefaultFieldsAttributeDict):
         'max_memory_kb',
         'rerunnable',
         'retrieve_list',  # a list of files or patterns to retrieve, with two
-        # possible formats: [ 'remotepath',  # just the name of the file to retrieve. Will be put in '.' of the repositorym with name os.path.split(item)[1]
-        # ['remotepath','localpath',depth]  ]
+        'retrieve_temporary_list', # a list of files or patters to retrieve that will only be kept during parsing and will be deleted after parsing is completed
+        # possible formats: ['remotepath',  # just the name of the file to retrieve. Will be put in '.' of the repository with name os.path.split(item)[1]
+        # ['remotepath','localpath', depth]]
         # second format will copy the remotepath file/folder to localpath.
```
Comment (giovannipizzi) [inline aiida/common/datastructures.py]: This comment belongs still to 'retrieve_list', whose description is not 'cut'

File: aiida/daemon/execmanager.py [hunk 2]
```diff
@@ -806,6 +771,7 @@ def retrieve_computed_for_authinfo(authinfo):
                         calc.pk, retrieved_files.dbnode.pk),
                                      extra=logger_extra)
                     retrieved_files.store()
+
```
Comment (giovannipizzi) [inline aiida/daemon/execmanager.py]: Add a comment that we are not storing the _temp files

Comment (giovannipizzi) [reply aiida/daemon/execmanager.py]: Thanks, fine for me. I'll approve this and let you merge or decide to put an explicit clear of the SandboxFolder for the temporary FolderData

**Comment by sphuber** [comment]
I have added a `TemplatereplacerParser` that does nothing but verify that, if a `retrieve_file_list` was specified in the `template` input node of the `TemplatereplacerCalculation`, that the temporary folder is passed to the `parse_with_retrieved` call by the execmanager. It seems to work, but I am not sure if this check is enough

**Pull request merged**